### PR TITLE
remove limits around arrays and clarify text - DOCSP-22050

### PR DESCRIPTION
**Pull request by MongoCaleb**
### Jira

https://jira.mongodb.org/browse/DOCSP-22050

### Staged Changes (Requires MongoDB Corp SSO)

https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-22050/reference/realm-query-language/
and
https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-22050/sync/data-access-patterns/flexible-sync/#eligible-field-types

### Note
Change to autobuilder/src/index.ts is a work-around for a borked autobuilder. Change should only be needed for the short term, but does no harm being in there for ever.

### CODE REVIEW

File: source/sync/data-access-patterns/flexible-sync.txt [hunk 1]
```diff
@@ -66,8 +66,8 @@ Eligible Field Types
 ~~~~~~~~~~~~~~~~~~~~
 
 Any top-level primitive field with a scalar type can be a queryable field. 
-Flexible Sync does not currently support embedded objects as 
-queryable fields. Flexible Sync queries only support arrays of primitives, not objects.
+Flexible Sync does not currently support embedded objects or arrays of 
+objects as queryable fields, but does support arrays of primitives.
```
Comment (mongodben) [inline source/sync/data-access-patterns/flexible-sync.txt:70]: i'd rec changing this whole paragraph to something like this for clarity: 

```
Flexible Sync only supports top-level primitive fields with a scalar type as queryable fields.
You can also include arrays of these primitives as queryable fields.
Flexible Sync does not currently support embedded objects or arrays of 
objects as queryable fields.
```

note: can't do suggestion mode b/c the suggestion deals w whole paragraph including unedited lines

Comment (mongodben) [reply source/sync/data-access-patterns/flexible-sync.txt:70]: this probably does for now, but the phrase `top-level primitive fields with a scalar type` seems needlessly opaque. 

maybe we could just list the supported types?

File: tools/autobuilder/src/index.ts [hunk 2]
```diff
@@ -6,7 +6,8 @@ import {
 } from "mongodb-stitch-server-sdk";
 
 // Add expected errors here.
-const expectedErrors: RegExp[] = [/(WARNING|ERROR)\(sdk\/java\/api.*/];
+const expectedErrors: RegExp[] = [/(WARNING|ERROR)\(sdk\/java\/api.*/,
+ /ERROR #98124  WEBPACK/];
```
Comment (mongodben) [inline tools/autobuilder/src/index.ts:10]: sorry to be pedantic, but i think this should be a separate PR since it's about a completely unrelated, build-related thing.

Comment (MongoCaleb) [reply tools/autobuilder/src/index.ts:10]: Perhaps. In this case, it is a blocking issue and hopefully temporary.

Comment (mongodben) [reply tools/autobuilder/src/index.ts:10]: ok, can you mention what this change is about in the PR description then?

**Review by mongodben — APPROVED** [review]
LGTM
